src/unit-tests/ostimer-test/ut_ostimer_timerio_test.c

```diff
@@ -319,7 +325,19 @@ void UT_os_timercreate_test()
         UT_OS_TEST_RESULT( testDesc, UTASSERT_CASETYPE_FAILURE);
 
     /*-----------------------------------------------------*/
-    testDesc = "#3 Name-taken";
+    testDesc = "#3 Unterminated String Test";
+    /* Reference https://github.com/nasa/osal/issues/88 for additional info */
+
+    strncpy( unterminatedStringTest, "12345678912345678900", OS_MAX_API_NAME + 1 );
+
+    if (OS_TimerCreate(&g_timerIds[2], unterminatedStringTest, &g_clkAccuracy, &UT_os_timercallback) ==
+        OS_ERR_NAME_TOO_LONG)
+        UT_OS_TEST_RESULT( testDesc, UTASSERT_CASETYPE_PASS);
```

Comment:
Correct.  The unterminatedStringTest variable is not an unterminated string.  It is exactly OS_MAX_API_NAME characters plus the terminating NULL.  The issue was that OS_TimerCreate was performing the following:
if (strlen(timer_name) > OS_MAX_API_NAME)
...return error value

such that if timer_name was exactly OS_MAX_API_NAME characters + terminating NULL(s), the check would pass - and then go on to perform the following:
strncpy(OS_timer_table[possible_tid].name, timer_name, OS_MAX_API_NAME);

which would knife the terminating NULL/create the unterminated string and probably result in the aforementioned segfault

It might be worth updating this to remove hard-coding the unterminatedStringTest variable length and make it so that the length is always OS_MAX_API_NAME?

Reply:
I like the memset.

Reply:
@jphickey are you suggesting a test name change?  Basically it's a name too long test, not an unterminated string test.  Name too long should be predictable behavior and can be protected for...

Reply:
Also remove the comment referring to the issue.  Our coding standards recommend avoiding this pattern.

Reply:
The supplied literal is 20 characters long, so this only works when `OS_MAX_API_NAME` is also set to 20.  If it is larger than 20, the test will still create a string that is only 20 characters, therefore the test will fail because it isn't too long anymore.  Conversely if `OS_MAX_API_NAME` is less than 20, it will actually create an unterminated string, which is worse.

Just memset as I described and you'll get a string of length `OS_MAX_API_NAME` which is the intent.  And call it something like `longStringTest` not `unterminatedStringTest`.  The string should never be unterminated.

Reply:
Also, another minor nitpick - shouldn't need a separate buffer for this.  The `UT_NAME_BUFF_SIZE` is already set to be slightly longer than the corresponding OSAL config, and this was intentended for testing the "name too long" outputs.  In ut_os_support.h:

```
#define UT_OS_NAME_BUFF_SIZE        (OS_MAX_API_NAME + 10)
```

So - can't we just use `tmpStr` buffer that already exists for this?  Just memset it with exactly `OS_MAX_API_NAME` chars and then set a NULL terminator at the end.

Reply:
Roger Wilco

Reply:
Code in commit fa833b4 looks good now